neo/Network/P2P/Payloads/Transaction.cs

```diff
@@ -56,72 +59,30 @@ public UInt256 Hash
 
         InventoryType IInventory.InventoryType => InventoryType.TX;
 
-        public int Size =>
-            sizeof(byte) +              //Version
-            sizeof(uint) +              //Nonce
-            Script.GetVarSize() +       //Script
-            Sender.Size +               //Sender
-            sizeof(long) +              //Gas
-            sizeof(long) +              //NetworkFee
-            sizeof(uint) +              //ValidUntilBlock
-            Attributes.GetVarSize() +   //Attributes
-            Witness.Size;               //Witnesses
+        public const int HeaderSize =
+            sizeof(byte) +  //Version
+            sizeof(uint) +  //Nonce
+            20 +            //Sender
+            sizeof(long) +  //Gas
+            sizeof(long) +  //NetworkFee
+            sizeof(uint);   //ValidUntilBlock
 
-        public void CalculateFees()
-        {
-            if (Sender is null) Sender = UInt160.Zero;
-            if (Attributes is null) Attributes = new TransactionAttribute[0];
-            if (Witness is null) Witness = new Witness
-            {
-                InvocationScript = new byte[65],
-                VerificationScript = new byte[39]
-            };
-            _hash = null;
-            long consumed;
-            using (ApplicationEngine engine = ApplicationEngine.Run(Script, this))
-            {
-                if (engine.State.HasFlag(VMState.FAULT))
-                    throw new InvalidOperationException();
-                consumed = engine.GasConsumed;
-            }
-            _hash = null;
-            long d = (long)NativeContract.GAS.Factor;
-            Gas = consumed - ApplicationEngine.GasFree;
-            if (Gas <= 0)
-            {
-                Gas = 0;
-            }
-            else
-            {
-                long remainder = Gas % d;
-                if (remainder == 0) return;
-                if (remainder > 0)
-                    Gas += d - remainder;
-                else
-                    Gas -= remainder;
-            }
-            using (Snapshot snapshot = Blockchain.Singleton.GetSnapshot())
-            {
-                long feeperbyte = NativeContract.Policy.GetFeePerByte(snapshot);
-                long fee = feeperbyte * Size;
-                if (fee > NetworkFee)
-                    NetworkFee = fee;
-            }
-        }
+        public int Size => HeaderSize +
+            Attributes.GetVarSize() +   //Attributes
+            Script.GetVarSize() +       //Script
```

Comment:
This is what mislead me @shargon ... if Script is part of the signed part, this part should be the header... so it should be counted on HeaderSize then.

Reply:
The idea is to make `HeaderSize` as a `const`.

Reply:
So, I.think its better to add scripthash (of script) in header, and take script out of header. In fact, it will be like a witness, but no verification script.

Reply:
No. It is an code optimization, not a logic change.